pkg/provisioner/sessions.go

```diff
@@ -0,0 +1,203 @@
+/*
+ *
+ *  MIT License
+ *
+ *  (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a
+ *  copy of this software and associated documentation files (the "Software"),
+ *  to deal in the Software without restriction, including without limitation
+ *  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ *  and/or sell copies of the Software, and to permit persons to whom the
+ *  Software is furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included
+ *  in all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ *  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ *  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ *  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *  OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+package provisioner
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// Session stores a TPM Provisioner session data.
+type Session struct {
+	xname    string
+	nodeType string
+	expiry   time.Time
+	step     int
+	nonce    string
+	reqData  string
+}
+
+var sessions = map[string]Session{}
+
+// createSessionCookie returns a basic session cookie.
+func createSessionCookie(xname string, nodeType string) (string, time.Time) {
+	token := uuid.NewString()
```

Comment:
IIRC, we're using both a JWT and a session. So, I'm guessing the session is simply to protect the challenge/response process. 128-bits seems reasonable as at least a state, provided the underlying lib is actually doing it's job w.r.t. pseudo-random behavior. Another future nit, might look at how the UUID behaves and if it should be replaced by another NONCE application, etc. Risk seems low and I'm guessing you did a bit of homework here.

Reply:
The session is there to track the multi-request process and to help prevent replays by preventing each step from being accepted twice.